R/zchunk_LB162.ag_prodchange_R_C_Y_GLU_irr.R

```diff
@@ -41,67 +45,460 @@ module_aglu_LB162.ag_prodchange_R_C_Y_GLU_irr_DISABLED <- function(command, ...)
+    # Repeat the resulting tibble for all SPEC_AG_PROD_YEARS, and join in the yield multipliers.
+    # This results in a table of harvested area and yield multipliers by CROSIT country and crop, glu,
+    # and irrigation for each SPEC_AG_PROD_YEARS year, based on LDS harvested area tables L151....
     #
+    # In this method, compositional shifts of CROSIT commodities within GCAM commodities do not
+    # translate to modified yields. Harvested area multipliers are 1 in all periods.
+    L162.ag_HA_ha_ctry_crop_irr %>%
+      group_by(CROSIT_ctry, CROSIT_crop, GLU, Irr_Rfd) %>%
+      summarise(HA = sum(HA)) %>%
+      na.omit() %>%
+      ungroup() %>%
+      # Filter to country-crop-irrigation combos in yield multiplier table
+      semi_join(select(L162.ag_Yieldmult_Rcrs_Ccrs_Y_irr, CROSIT_ctry, CROSIT_crop, Irr_Rfd),
+                by = c("CROSIT_ctry", "CROSIT_crop", "Irr_Rfd")) %>%
+      # repeat for all years in SPEC_AG_PROD_YEARS and join Yield multipliers for each year
+      repeat_add_columns(tibble::tibble(year = SPEC_AG_PROD_YEARS)) %>%
+      left_join_error_no_match(L162.ag_Yieldmult_Rcrs_Ccrs_Y_irr,
+                               by = c("CROSIT_ctry", "CROSIT_crop", "Irr_Rfd", "year")) ->
+      L162.ag_HA_ha_Rcrs_Ccrs_Ysy_GLU_irr
+
+
+
+    # Calculating the adjusted production / harvested area in each time period, for each GCAM region / commodity / GLU.
+    # Starting from full GTAP table for composite regions and commodities in the CROSIT database, L162.ag_HA_ha_ctry_crop_irr,
+    # repeating for all years in SPEC_AG_PROD_YEARS.
+    # Then join in yield multipliers from CROSIT L162.ag_HA_ha_Rcrs_Ccrs_Ysy_GLU_irr, GCAM region and GCAM commodity information.
+    # Calculate modified production, Prod_mod = base year harvested area HA * yearly yield multipliers Mult. Production and
+    # Harvested area are then aggregated to the GCAM region - commodity level so that Aggregated Yield can be correctly calculated.
+    # The YieldRatio = Prod_mod/HA is then calculated and output for each GCAM region-commodity-glu-irrigation-year.
+    #
+    # There is a small error in the old data system. The intention is to match in multipliers by CROSIT_ctry, CROSIT_crop, irrigation,
+    # and year. The old DS only does so by CROSIT_ctry, CROSIT_crop, and year.
+
+    if(OLD_DATA_SYSTEM_BEHAVIOR) {
+      # preprocess table of multipliers before joining, restricting to the CROSIT country-crop-glu-irrigation present in
+      # the full GTAP table,  L162.ag_HA_ha_ctry_crop_irr
+      L162.ag_HA_ha_Rcrs_Ccrs_Ysy_GLU_irr %>%
+        select(CROSIT_ctry, CROSIT_crop, year, Mult) %>%
+        semi_join(L162.ag_HA_ha_ctry_crop_irr, by = c("CROSIT_ctry", "CROSIT_crop")) %>%
+        dplyr::distinct() ->
+        CROSIT_mult
+
+      L162.ag_HA_ha_ctry_crop_irr %>%
+        na.omit() %>%
+        repeat_add_columns(tibble::tibble(year = SPEC_AG_PROD_YEARS)) %>% # agree fine to here; it's the next join
```

Comment:
same debugging comment that I missed. I should start doing them with `##` so I can just use find to make sure I remove all. There are very possibly other similar comments and I will go through line by line to make sure they're all removed